src/main/java/core/basesyntax/service/impl/TransactionProcessor.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.TransactionParser;
+import java.util.List;
+
+public class TransactionProcessor {
+    private final OperationStrategy operationStrategy;
+    private final TransactionParser transactionParser;
+
+    public TransactionProcessor(OperationStrategy operationStrategy,
+                                TransactionParser transactionParser) {
+        this.operationStrategy = operationStrategy;
+        this.transactionParser = transactionParser;
+    }
+
+    public void processTransactions(List<String> inputData) {
+        inputData.stream()
+                .map(transactionParser::parseLine)
+                .forEach(transaction -> {
+                    OperationHandler handler = operationStrategy
+                            .getHandler(transaction.getOperation());
+                    Fruit fruit = new Fruit(transaction.getFruitName());
+                    handler.apply(fruit, transaction.getQuantity());
+                });
+    }
+}
```

Comment:
It's good to see that you've created a `TransactionProcessor` class to handle the processing of transactions. However, please make sure that this class is called properly in the `main()` method of your application, and that it's not tightly coupled with other classes according to the SOLID principles. Each service should be independent and called in the right order in `main()` method.